karrot/groups/serializers.py

```diff
@@ -371,11 +370,22 @@ def get_application_questions(self, group) -> str:
         return group.get_application_questions_or_default()
 
     def get_member_count(self, group) -> int:
+        annotated = getattr(group, 'member_count', None)
+        if annotated is not None:
+            return annotated
+
         return group.members.count()
 
     def get_is_member(self, group) -> bool:
         user = self.context['request'].user if 'request' in self.context else None
-        return user in group.members.all() if user else False
+        if not user or user.is_anonymous:
+            return False
+
+        annotated = getattr(group, 'is_user_member', None)
+        if annotated is not None:
+            return annotated
+
+        return user in group.members.all()
```

Comment:
I wondered if this would be easier to read:

```py
try:
  return group.is_user_member
except AttributeError:
  return user in group.members.all()
```

Reply:
I find this way cleaner looking (and also used elsewhere):

```py
if hasattr(group, 'is_user_member'):
  return group.is_user_member

return user in group.members.all()
```